module-services/service-desktop/Sync.cpp

```diff
@@ -0,0 +1,180 @@
+}
+
+bool Sync::RemoveSyncDir(const std::filesystem::path &path)
+{
+    /* prepare directories */
+    if (std::filesystem::is_directory(path)) {
+        LOG_INFO("Removing sync directory %s...", path.c_str());
+        std::error_code errorCode;
+
+        if (std::filesystem::remove_all(path, errorCode) == 0) {
+            LOG_ERROR("Removing sync directory %s failed, error: %d.", path.c_str(), errorCode.value());
+            return false;
+        }
+    }
+
+    return true;
+}
+
+bool Sync::CreateSyncDir(const std::filesystem::path &path)
+{
+    LOG_INFO("Creating sync directory %s...", path.c_str());
+    if (std::filesystem::exists(path)) {
+        LOG_ERROR("Sync directory already exists!");
+        return false;
+    }
+    std::error_code errorCode;
+    if (!std::filesystem::create_directories(path, errorCode)) {
+        LOG_ERROR("Failed to create directory: %s, error: %d", path.c_str(), errorCode.value());
+        return false;
+    }
+    return true;
+}
+
+bool Sync::PackSyncFiles(const std::filesystem::path &path)
+{
+    if (std::filesystem::is_empty(path)) {
+        LOG_ERROR("Sync dir is empty, quitting...");
+        return false;
+    }
+
+    auto isTarFileOpen                = false;
+    std::filesystem::path tarFilePath = (purefs::dir::getSyncPackagePath() / path.filename());
+    mtar_t tarFile;
+
+    auto cleanup = gsl::finally([&isTarFileOpen, &tarFile]() {
+        if (isTarFileOpen) {
+            mtar_close(&tarFile);
+        }
+    });
+
+    LOG_INFO("Opening tar %s file...", tarFilePath.c_str());
+
+    int ret = mtar_open(&tarFile, tarFilePath.c_str(), "w");
+    if (ret != MTAR_ESUCCESS) {
+        LOG_ERROR("Opening tar file failed, quitting...");
+        return false;
+    }
+    isTarFileOpen = true;
+
+    auto fileStreamBuffer = std::make_unique<char[]>(purefs::buffer::tar_buf);
+    auto tarStreamBuffer  = std::make_unique<char[]>(purefs::buffer::tar_buf);
+    setvbuf(tarFile.stream, tarStreamBuffer.get(), _IOFBF, purefs::buffer::tar_buf);
+    std::error_code errorCode;
+
+    for (auto &direntry : std::filesystem::directory_iterator(path)) {
+        if (!isValidDirentry(direntry)) {
+            continue;
+        }
+
+        LOG_INFO("Archiving file ...");
+        LOG_DEBUG("Writing tar header ...");
+
+        if (mtar_write_file_header(&tarFile,
+                                   direntry.path().filename().c_str(),
+                                   static_cast<unsigned>(std::filesystem::file_size(direntry))) != MTAR_ESUCCESS) {
+            LOG_ERROR("Writing tar header failed");
+            return false;
+        }
+
+        uintmax_t filesize = std::filesystem::file_size(direntry.path(), errorCode);
+        if (errorCode) {
+            LOG_ERROR("Failed to get size for file %s, error: %d", direntry.path().c_str(), errorCode.value());
+            return false;
+        }
+        auto loopCount = (filesize / purefs::buffer::tar_buf) + 1u;
```

Comment:
`const`

Reply:
Would it make sense to defined it as `uintmax_t` instead of `auto`?

Reply:
I agree about `const`, but I do not see any benefit of changing `auto` to `uintmax_t`.

Reply:
I personally prefer to use explicit types if they are simple and non-generic because it gives more information while reading the code. But I'm ok with `auto` if you prefer it.